torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -0,0 +1,93 @@
+#include "torch/csrc/jit/passes/constant_propagation.h"
+#include "torch/csrc/jit/passes/dead_code_elimination.h"
+#include "torch/csrc/utils/functional.h"
+#include "torch/csrc/jit/interpreter.h"
+
+namespace torch { namespace jit {
+
+std::unordered_set<Symbol> skip_list = {
+  //FIXME If & Loop require special casing because they cannot be run as a
+  //single node.
+  prim::If,
+  prim::Loop,
+  //FIXME Same problem as in DCE - cpp & python PythonOp and CppOp should be
+  //FIXME treated as having side effects but ONNX depends on them being removed
+  prim::Print,
+ };
+
+std::vector<at::Tensor> runNode(Node *n) {
+  auto graph = std::make_shared<Graph>();
+  auto block = graph->block();
+  //very similar to Block::cloneFrom, but here we are cloning just the node,
+  //and not recursively
+  std::unordered_map<Value*, Value*> local_map;
+  auto env = [&](Value * v) {
+    auto it = local_map.find(v);
+    if(it != local_map.end())
+      return it->second;
+    barf("Encountered a use of a value not in scope");
```

Comment:
I think the canonical way to do those things is usually (has a benefit of e.g. providing a line number and the file)
```cpp
JIT_ASSERTM(it != local_map.end(), "Encountered a use of a value not in scope")
```